servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java

```diff
@@ -3829,9 +3829,32 @@ public void deleteRole(ResourceContext ctx, String domainName, String roleName,
             throw ZMSUtils.requestError("deleteRole: admin role cannot be deleted", caller);
         }
 
+        /* we are not going to allow any user to delete
+         * role that related to any assertion of any policy */
+
+        Policies policies = getPolicies(ctx, domainName, true, true);
+
+        // size > 1 because of the admin policy
+        if (policies.getList().size() > 1) {
+            validateRoleNotAssociatedToPolicy(policies.getList(), roleName, domainName, caller);
+        }
+
         dbService.executeDeleteRole(ctx, domainName, roleName, auditRef, caller);
     }
 
```

Comment:
In this function i don't check the case that the role that is about to delete is the last one that match to an assertion with a wild card. Do it?